### PR TITLE
GCP CRUD bucket

### PR DESCRIPTION
**Pull request by vineela1999**
This PR includes CRUD bucket code of GCP cloud
**multipart upload of objects feature(huge objects) is not tested**

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
> /kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
kind NOT-TESTED

**Test Report**:

**Special notes for your reviewer**:

### CODE REVIEW

File: s3/pkg/datastore/gcp/gcp.go [hunk 1]
```diff
@@ -46,9 +46,66 @@ type GcsAdapter struct {
...
+	Credentials := credentials.NewStaticCredentials(ad.backend.Access, ad.backend.Security, "")
+	configuration := &aws.Config{
+		Region:      Region,
+		Endpoint:    Endpoint,
+		Credentials: Credentials,
+	}
+
+	svc := awss3.New(session.New(configuration))
+
+	input1 := &awss3.CreateBucketInput{
```
Comment (kumarashit) [inline s3/pkg/datastore/gcp/gcp.go:91]: Maybe a better variable name than 'input1'.

File: s3/pkg/datastore/gcp/gcp.go [hunk 2]
```diff
@@ -46,9 +46,66 @@ type GcsAdapter struct {
...
+	input := &awss3.DeleteBucketInput{
+		Bucket: aws.String(in.Name),
+	}
+
+	result, err := svc.DeleteBucketWithContext(ctx, input)
+	if err != nil {
+		log.Error("failed to delete bucket due to %s:", err)
+		return err
+	}
+	log.Debug("The bucker deletion is successful in gcp:%s",result)
```
Comment (kumarashit) [inline s3/pkg/datastore/gcp/gcp.go:71]: typo:bucker->bucket

File: s3/pkg/datastore/gcp/gcp.go [hunk 3]
```diff
@@ -46,9 +46,66 @@ type GcsAdapter struct {
...
+		Credentials: Credentials,
+	}
+
+	svc := awss3.New(session.New(configuration))
+
+	input := &awss3.DeleteBucketInput{
+		Bucket: aws.String(in.Name),
+	}
+
+	result, err := svc.DeleteBucketWithContext(ctx, input)
```
Comment (kumarashit) [inline s3/pkg/datastore/gcp/gcp.go:66]: Put a comment as in why DeleteBucketWithContext is used nit simple DeletBucket

File: s3/pkg/datastore/gcp/gcp.go (hunk 1 above)
Comment (vineela1999) [reply s3/pkg/datastore/gcp/gcp.go]: changed

File: s3/pkg/datastore/gcp/gcp.go (hunk 2 above)
Comment (vineela1999) [reply s3/pkg/datastore/gcp/gcp.go]: changed

File: s3/pkg/datastore/gcp/gcp.go (hunk 3 above)
Comment (vineela1999) [reply s3/pkg/datastore/gcp/gcp.go:66]: included

**Review by kumarashit — APPROVED** [review]
LGTM

**Comment by PravinRanjan10** [comment]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**